selectors/drafts.ts

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {createSelector} from 'reselect';
+
+import {getMyChannels} from 'mattermost-redux/selectors/entities/channels';
+import {GlobalState} from 'types/store';
+import {PostDraft} from 'types/store/rhs';
+
+import {StoragePrefixes} from 'utils/constants';
+
+type Info = {
+    id: string;
+    type: 'channel' | 'thread';
+}
+
+export type Draft = Info & {
+    key: keyof GlobalState['storage']['storage'];
+    value: PostDraft;
+    timestamp: Date;
+};
+
+export type DraftSelector = (state: GlobalState) => Draft[];
+export type DraftCountSelector = (state: GlobalState) => number;
+
+function getInfoFromKey(key: string, prefix: string): Info {
+    const keyArr = key.split('_');
+    if (prefix === StoragePrefixes.DRAFT) {
+        return {
+            id: keyArr[1],
+            type: 'channel',
+        };
+    }
+
+    return {
+        id: keyArr[2],
+        type: 'thread',
+    };
+}
+
+export function makeGetDraftsByPrefix(prefix: string): DraftSelector {
+    return createSelector(
+        'makeGetDraftsByPrefix',
+        (state: GlobalState) => state.storage?.storage,
+        (storage) => {
+            if (!storage) {
+                return [];
+            }
+
+            return Object.keys(storage).flatMap((key) => {
```

Comment:
Huh, I didn't know `.flatMap` existed. Is that actually necessary here though? I'd assume `Object.keys` would always return a flat array anyway.

Reply:
I recently found out it seems that you can map and filter in the same loop,
filtering is done by returning an empty array :tada:
The mobile app uses it extensively.

Reply:
Oh, I see. I thought it flattened the input and then did a map so it could be used for stuff like `[[1, 2], 3, [4, 5, 6]].flatMap`, but flattening after the map makes much more sense. Using it to filter and map at the same time feels a bit counterintuitive, but it's useful enough that I can ignore that.